src/project/internal/audacityproject.cpp

```diff
@@ -97,6 +99,28 @@ muse::Ret Audacity4Project::doLoad(const io::path_t& path, bool forceMode, const
     UNUSED(forceMode);
     UNUSED(format);
 
+    // check file access
+    if (!io::FileInfo::exists(path)) {
+        LOGE() << "file does not exist at path: \"" << path << "\"";
+        return make_ret(Err::ProjectFileNotFound, path);
+    }
+
+    // check for file permissions
+    {
+        io::File file(path);
+        if (!file.open(io::IODevice::ReadOnly)) {
+            LOGE() << "failed open file (Can't Read): " << path;
+            return make_ret(Err::ProjectFileIsReadProtected, path);
+        }
+        file.close();
+
+        if (!file.open(io::IODevice::WriteOnly)) {
```

Comment:
I don't think we need to check for this particular case here. Not being able to open the file in "w" mode is one of the errors produced by ProjectFileIO.cpp:848. Therefore, we should either handle all of them or not bother.